### DISCUSSION TITLE
Allow customizing content of QSelect dropdown

### DISCUSSION THREAD

**Discussion by thevladisss** [quasarframework/quasar#17260] [Ideas / Proposals]
**Problem**
For QSelect element there is no legal way to customize the content rendered within the dropdown that is to say place custom content into QMenu that shows options of QSelect.  QSelect does not expose slot to allow explicitly passing content to QMenu that is rendered by triggering open of QSelect  and only allows cutsomizing individual options.

**Solution**
Expose slot that would allow accessing options prop data and rendering custom content within QMenu that is rendered by QSelect

**Comment by ivanjaros** [comment]
Use your own q-input + q-menu then. QSelect is exactly that - a select.

**Reply by thevladisss** [reply]
> Use your own q-input + q-menu then. QSelect is exactly that - a select.

That does not mean I want to change it to be something else. There just should be an opportunity to change contents

**Reply by ivanjaros** [reply]
maybe you can just use the option slot?

**Reply by thevladisss** [reply]
> maybe you can just use the option slot?

That does not quite imply customizing contents of the dropdown. I do beleve that the thing I suggest makes sense

**Comment by yusufkandemir** [comment]
You should be able to use `#before-options` and `#after-options` to add the customizations you want. If it doesn't work, please share an example use case in which adding this feature would have made possible.

**Reply by danstooamerican** [reply]
It would be possible to display options with groups and subheadings.

```
Location A

- Option 1
- Option 2
- Option 3

Location B

- Option 4
- Option 5
```

Only the options are selectable and the headers (i.e., Location A) are inserted. This is not possible with before-options and after-options.

**Reply by stefanvanherwijnen** [reply]
Use the `options` slot to customize the (display of) options:

https://codepen.io/stefanvh/pen/XJmgrax
